pkg/csi/service/vanilla/controller.go

```diff
@@ -736,12 +736,12 @@ func (c *controller) ControllerUnpublishVolume(ctx context.Context, req *csi.Con
 		if err != nil {
 			msg := fmt.Sprintf("QueryVolume failed for volumeID: %q. %+v", req.VolumeId, err.Error())
 			log.Error(msg)
-			return nil, status.Error(codes.Internal, msg)
+			return nil, status.Error(codes.Aborted, msg)
```

Comment:
why are you returning `codes.Aborted` when operation fails.

error code Aborted means

This error code should be used for "Operation pending for volume"

> Indicates that there is already an operation pending for the specified volume. In general the Cluster Orchestrator (CO) is responsible for ensuring that there is no more than one call "in-flight" per volume at a given time. However, in some circumstances, the CO MAY lose state (for example when the CO crashes and restarts), and MAY issue multiple calls simultaneously for the same volume. The Plugin, SHOULD handle this as gracefully as possible, and MAY return this error code to reject secondary calls.

Refer CSI Spec - https://github.com/container-storage-interface/spec/blob/master/spec.md#error-scheme